### ISSUE TITLE
Option to remember window layout from last session

### ISSUE THREAD

**Issue by herder** [pop-os/shell#735]
As many have sadi: Thank you for this extension! It was the last piece missing in the already fantastic Pop desktop :bow:  

I have one feature suggestion:

I have a lot of windows open during a normal session, some stacked and some not. When I logout or have to restart the computer it always takes a while to get my windows in order again.

Would it be possible to add a feature to remember the last window layout, something like OSX does, and restore it on next login? I'm not sure if it possible and how much support there is in Gnome for getting data on windows and positions, but it would be a very useful feature.

**Comment by hoebbelsB** [comment]
I would love to see such a feature aswell. In my opinion https://github.com/material-shell/material-shell does this very good! If it only had something like the stacking mode :(

**Comment by pvannierop** [comment]
I second this request. Is there a command line interface to the window manager so that we can write a script that sets up all windows on different desktops?

**Comment by varac** [comment]
Related, but different: #413

**Comment by funnylookinhat** [comment]
This feature would be amazing when switching between different monitor configurations as well.  When I have my laptop unplugged vs. plugged into monitors.

**Comment by miellis** [comment]
bump for this feature!

**Comment by anzanhoshin** [comment]
Indeed. This would be perfect.

**Comment by agustinsacco** [comment]
Super down for this. +1

**Comment by osscar** [comment]
super duper down +2

**Comment by zippoxer** [comment]
Please!

**Comment by awicenec** [comment]
Yes, please! Mac OSX also has profiles, which allows saving such information depending on where you are (e.g. home or work)

**Comment by zsilet** [comment]
I too would love to see this implemented. Currently I am having to rearrange my windows multiple times a day because (due to an unrelated issue) when my main (43" TV) and side (24" monitor) come out of 'sleep', they do so with slightly different timing, causing my laptop to momentarily rearrange all the windows from multiple virtual desktops to the side monitor. When the main monitor comes back online, all windows are bunched up in one place.

**Comment by xxSkyy** [comment]
Bump

**Comment by eliot-akira** [comment]
There's a GNOME shell extension called [Another Window Session Manager](https://github.com/nlpsuge/gnome-shell-extension-another-window-session-manager), which can save and restore sessions with a set of open applications and their positions/layout. (It can be installed from [here](https://extensions.gnome.org/extension/4709/another-window-session-manager/).)

It also has an option to restore a session on startup.

I've only been using it for a day or so, but I think this solves the issue for me. It would be nice if this were a built-in feature of Pop Shell.

Edit: It seems the window session manager is not quite compatible. It crashed a couple times so I uninstalled it.

**Comment by bastoune** [comment]
WOULD LOVE IT !

**Comment by ykill** [comment]
I recommend the above also. [Another Window Session Manager.](https://github.com/nlpsuge/gnome-shell-extension-another-window-session-manager) works for me using tumbleweed with [Pop Shell](https://github.com/pop-os/shell) which I compile from source. _**EDIT:**_ I ended up uninstalling, because it took so long to set up windows and had some bugs.

**Comment by r14n** [comment]
It would be awesome to be able to assign applications to a particular tile or stack.  For example I like to divide my screen into 2 rows and 3 columns, with the middle column being the widest.  I always want to have Obsidian in the top left, Sublime top middle, a certain Chrome window (whatsapp) top right, stack of terminal windows lower left, "main" Chrome window bottom middle, and a certain Chrome window (homeassistant) bottom right.

If I could persist my tiles/stacks, and assign apps or window titles since I use Chrome for so many different ones) to each "zone", I would be so set.

**Comment by matthieucx** [comment]
I would be very interested in such a feature!

**Comment by kevgilmore** [comment]
It would also be useful to remember layout when enabling/disabling pop shell. 

For example currently, if I have several windows tiled and double press `ctrl+y` it will reorder the windows as it can't remember the layout.

**Comment by matthieucx** [comment]
Would also recommend [Another Windows Session Manager](https://github.com/nlpsuge/gnome-shell-extension-another-window-session-manager) for the original request.  (I installed it through the [GNOME Shell Extension Manager](https://flathub.org/apps/com.mattjakeman.ExtensionManager))

Works flawlessly on Pop-OS 22.04. I've been testing it for a couple of weeks, it only takes a few seconds to launch and reorder 7 windows between two screens and two workspaces.

**Comment by PHHENS** [comment]
This feature is a must have which should be integrated to gnome (no need to use extensions).
Tried several extensions (AWSM, smart automove ...) but none works really reliably...

**Comment by r14n** [comment]
> Would also recommend [Another Windows Session Manager](https://github.com/nlpsuge/gnome-shell-extension-another-window-session-manager) for the original request. (I installed it through the [GNOME Shell Extension Manager](https://flathub.org/apps/com.mattjakeman.ExtensionManager))
> 
> Works flawlessly on Pop-OS 22.04. I've been testing it for a couple of weeks, it only takes a few seconds to launch and reorder 7 windows between two screens and two workspaces.

When you say it works flawlessly I am assuming that is without pop's window tiling?  Because when I restore the windows are not in order if I use tiling.
